### PR TITLE
Updates to build and run on Orion Rocky 9

### LINKED ISSUE #754: Update Orion build to Rocky 9
Received the following from RDHPCS Management

> Orion’s Operating System (OS) and software stack is scheduled to be upgraded during a two day downtime, starting on Wednesday, June 12th and going through Thursday, June 13th.  The OS on Orion will be upgraded from CentOS 7 to Rocky 9, another derivative of Red Hat Linux.

This issue is opened to document the updating of `modulefiles/gsi_orion.intel.lua` to Rocky 9

### PR DESCRIPTION
**Pull request by RussTreadon-NOAA**
**Description**
This PR updates NOAA-EMC/GSI to build and run on Orion Rocky 9. 

Resolves #754

**Type of change**
- [x]  Maintenance

**How Has This Been Tested?**
Install on Orion and run ctests with [results](https://github.com/NOAA-EMC/GSI/issues/754#issuecomment-2191937540) (all tests _Passed_) posted in issue #754.
  
**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] New and existing tests pass with my changes

### CODE REVIEW

**Comment by RussTreadon-NOAA** [comment]
Open PR in _draft_ mode given increased wall times observed for `gsi.x` and `enkf.x` when run on Orion Rocky 9.

**Comment by RussTreadon-NOAA** [comment]
Slowness of `gsi.x` and `enkf.x` on Orion Rocky 9 remains unexplained but will change this PR to _Ready for review_ to invite feedback on proposed changes.

File: regression/regression_var.sh [hunk 1]
```diff
@@ -40,8 +40,13 @@ elif [[ -d /ncrc ]]; then # Gaea
...
-  export machine="Orion"
+elif [[ -d /work ]]; then # Orion or Hercules
+  mount=$(findmnt -n -o SOURCE /home)
+  if [[ ${mount} =~ "hercules" ]]; then
+    export machine="Hercules"
+  else
+    export machine="Orion"
+  fi
 elif [[ -d /work && $(hostname) =~ "hercules" ]]; then # Hercules
   export machine="Hercules"
```
Comment (DavidHuber-NOAA) [inline regression/regression_var.sh]: This should probably be removed.

Comment (RussTreadon-NOAA) [reply regression/regression_var.sh]: Good catch!  Unnecessary hercules check removed at 9c2cbf81.

**Review by DavidHuber-NOAA — APPROVED** [review]
Looks good to me.  Hopefully between spack-stack developers and Orion admins we can sort out the slowdown on Orion.

**Comment by RussTreadon-NOAA** [comment]
Agreed, @DavidHuber-NOAA !   The `gsi.x` and `enkf.x` slowdown on Orion Rocky 9 does not make sense to me.

**Comment by aerorahul** [comment]
@RussTreadon-NOAA 
Can this PR be merged with the knowledge that GSI is running in degraded status on Orion?
This will allow the global-workflow to proceed for Orion+Rocky8.
When the source of the degradation is identified on Orion, we can update the submodule pointer for Orion+Rocky8

Tagging @CatherineThomas-NOAA for awareness and discussions on the GFSv17 project tag-up.

**Comment by RussTreadon-NOAA** [comment]
@aerorahul , let me check with the GSI Review team

@ShunLiu-NOAA , @CoryMartin-NOAA , and @hu5970 :  Are we OK merging this PR into GSI `develop` even though ctests show `gsi.x` and `enkf.x` run approximately 2x slower on Orion Rocky 9 and Orion Centos 7?   @aerorahul explains [above](https://github.com/NOAA-EMC/GSI/pull/764#issuecomment-2197535185) why this question is being asked.

I'm reluctant to merge since do so may lessen the urgency of addressing the 2x slowdown.  That said, we don't want NOAA-EMC/GSI to become the roadblock for completion of the g-w transition to Orion Rocky 9 (see issue [#2694](https://github.com/NOAA-EMC/global-workflow/issues/2694))

**Comment by CoryMartin-NOAA** [comment]
@RussTreadon-NOAA I share your concerns but I don't think we have a choice. It's either "runs slow" or "not at all", so I think we have to go with the former.

**Comment by aerorahul** [comment]
> @aerorahul , let me check with the GSI Review team
> 
> @ShunLiu-NOAA , @CoryMartin-NOAA , and @hu5970 : Are we OK merging this PR into GSI `develop` even though ctests show `gsi.x` and `enkf.x` run approximately 2x slower on Orion Rocky 9 and Orion Centos 7? @aerorahul explains [above](https://github.com/NOAA-EMC/GSI/pull/764#issuecomment-2197535185) why this question is being asked.
> 
> I'm reluctant to merge since do so may lessen the urgency of addressing the 2x slowdown. That said, we don't want NOAA-EMC/GSI to become the roadblock for completion of the g-w transition to Orion Rocky 9 (see issue [#2694](https://github.com/NOAA-EMC/global-workflow/issues/2694))

Just open another issue to report the performance degradation on Orion after the upgrade and follow the development there.

**Review by CoryMartin-NOAA — APPROVED** [review]

**Comment by RussTreadon-NOAA** [comment]
Actions already taken reporting `gsi.x` and `enkf.x` slowdown on Orion Rocky-9:
 - open ticket RDHPCS ticket #2024062754000098 with Orion Helpdesk 
 - open spack-stack issue [#1166](https://github.com/JCSDA/spack-stack/issues/1166)

Someone will need to follow up on the ticket and issue.  doing so will likely require trying various things until the problem is resolved.

**Pull request merged**